bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2338,33 +2338,37 @@ private static boolean checkIsLikeArrayType(Object sourceValue, BArrayType targe
         if (source.getType().getTag() == TypeTags.ARRAY_TAG) {
             Type sourceElementType = ((BArrayType) source.getType()).getElementType();
             if (isValueType(sourceElementType)) {
-                boolean isType = checkIsType(sourceElementType, targetTypeElementType, new ArrayList<>());
 
-                if (isType || !allowNumericConversion || !isNumericType(sourceElementType)) {
-                    return isType;
-                }
-
-                if (isNumericType(targetTypeElementType)) {
+                if (checkIsType(sourceElementType, targetTypeElementType, new ArrayList<>())) {
                     return true;
                 }
 
-                if (targetTypeElementType.getTag() != TypeTags.UNION_TAG) {
-                    return false;
-                }
+                if (allowNumericConversion && isNumericType(sourceElementType)) {
+                    if (isNumericType(targetTypeElementType)) {
+                        return true;
+                    }
 
-                List<Type> targetNumericTypes = new ArrayList<>();
-                for (Type memType : ((BUnionType) targetTypeElementType).getMemberTypes()) {
-                    if (isNumericType(memType) && !targetNumericTypes.contains(memType)) {
-                        targetNumericTypes.add(memType);
+                    if (targetTypeElementType.getTag() != TypeTags.UNION_TAG) {
+                        return false;
```

Comment:
Shall we check the codecov comment and add a test that covers this scenario if we don't have one already?

Reply:
A test case is already there in `valuelib_test.bal`, which will cover this. https://github.com/ballerina-platform/ballerina-lang/blob/af74458a5e9ac20edd4f44939b6908e38fc68bc0/langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal#L661